src/NServiceBus.Core/Hosting/StartupDiagnostics/HostStartupDiagnosticsWriter.cs

```diff
@@ -58,6 +49,32 @@ await diagnosticsWriter(data)
             }
         }
 
+        List<StartupDiagnosticEntries.StartupDiagnosticEntry> DeduplicateEntries(List<StartupDiagnosticEntries.StartupDiagnosticEntry> entries)
+        {
+            var countMap = new Dictionary<string, int>(StringComparer.OrdinalIgnoreCase);
+            var newList = new List<StartupDiagnosticEntries.StartupDiagnosticEntry>();
```

Comment:
Can probably use `yield return` to avoid creating a new `List<>` instance.

Reply:
Done.